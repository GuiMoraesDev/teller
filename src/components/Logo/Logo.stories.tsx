import React from 'react';

import { ComponentMeta, ComponentStory } from '@storybook/react';

import Logo from '.';

const Storie = {
	title: 'Components/Logo',
	component: Logo,
} as ComponentMeta<typeof Logo>;

export default Storie;

const Template: ComponentStory<typeof Logo> = (args) => <Logo {...args} />;

export const DefaultTypeIcon = Template.bind({});

DefaultTypeIcon.args = {
	type: 'icon',
};

export const DefaultTypeLogo = Template.bind({});

DefaultTypeLogo.args = {
	type: 'logo',
};
