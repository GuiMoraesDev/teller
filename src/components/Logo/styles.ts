import styled from 'styled-components';

export const Logo = styled.img`
	height: 100%;
	max-height: 20rem;
`;
