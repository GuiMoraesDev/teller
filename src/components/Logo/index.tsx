import * as Styles from './styles';

interface Props {
	type?: 'icon' | 'logo';
}

const Logo = ({ type = 'icon' }: Props): JSX.Element => {
	return (
		<Styles.Logo
			src={
				type === 'icon'
					? 'img/icon.svg'
					: 'img/logo.svg'
			}
			alt={
				type === 'icon'
					? "Teller logo it's a microphone with sound waves around it and looks like a face blinking eyes"
					: "Teller logo it's a microphone with sound waves around it and looks like a face blinking eyes and Teller aside"
			}
		/>
	);
};

export default Logo;
