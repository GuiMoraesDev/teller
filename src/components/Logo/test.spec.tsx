import React from 'react';

import { render } from 'tests/test.utils';

import Logo from './index';

describe('Logo component', () => {
	it('should render correctly type icon', () => {
		const { container } = render(<Logo type="icon" />);

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should render correctly type logo', () => {
		const { container } = render(<Logo type="logo" />);

		expect(container.firstChild).toMatchSnapshot();
	});
});
