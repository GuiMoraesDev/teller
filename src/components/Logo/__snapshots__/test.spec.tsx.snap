// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component should render correctly type icon 1`] = `
.c0 {
  height: 100%;
  max-height: 20rem;
}

<img
  alt="Teller logo it's a microphone with sound waves around it and looks like a face blinking eyes"
  class="c0"
  src="img/icon.svg"
/>
`;

exports[`Logo component should render correctly type logo 1`] = `
.c0 {
  height: 100%;
  max-height: 20rem;
}

<img
  alt="Teller logo it's a microphone with sound waves around it and looks like a face blinking eyes and Teller aside"
  class="c0"
  src="img/logo.svg"
/>
`;
